src/main/kotlin/tech/thatgravyboat/skyblockpv/screens/tabs/combat/BestiaryScreen.kt

```diff
@@ -0,0 +1,152 @@
+            246,
+        )
+
+        val buttonContainer = carousel!!.getIcons { icons.map { Displays.item(it, showTooltip = true) } }
+
+        widget(
+            LayoutBuild.vertical(5) {
+                widget(buttonContainer.centerHorizontally(uiWidth))
+                widget(carousel!!.centerHorizontally(uiWidth))
+            }.asScrollable(uiWidth, uiHeight),
+        )
+    }
+
+    private fun getCategories(): Map<ItemStack, Display> = BestiaryCodecs.data?.categories?.map { (_, v) ->
+        Either.unwrap(
+            v.mapBoth(
+                { it.icon.getItem(it.name) to it.getCategory() }, // Simple
+                { it.icon.getItem(it.name) to it.getCategory() }, // Complex
+            ),
+        )
+    }?.toMap() ?: emptyMap()
+
+    private fun BestiaryCategoryEntry.getCategory() = mobs.map { it.getItem() }.format(MOBS_PER_ROW_SIMPLE)
+
+    private fun ComplexBestiaryCategoryEntry.getCategory() = subcategories.flatMap { it.value.mobs.map { it.getItem() } }.format(MOBS_PER_ROW_COMPLEX)
+
+    private fun List<Display>.format(mobsPerRow: Int) = toMutableList()
+        .rightPad(mobsPerRow * 2, Displays.empty(16, 16))
+        .map { Displays.padding(2, it) }
+        .chunked(mobsPerRow)
+        .let {
+            Displays.inventoryBackground(
+                mobsPerRow, it.size,
+                Displays.padding(2, it.asTable()),
+            )
+        }
+
+    private fun BestiaryMobEntry.getItem(): Display {
+        val kills = profile?.bestiaryData?.filter { mobs.contains(it.mobId) }?.sumOf { it.kills } ?: 0
+        val fullBracket = BestiaryCodecs.data?.brackets?.get(bracket) ?: emptyList()
+        val maxLevel = fullBracket.indexOf(cap) + 1
+        val bracket = fullBracket.take(maxLevel)
+        val requiredKills = bracket.lastOrNull() ?: 0
+        val currentLevel = bracket.indexOfLast { kills >= it } + 1
+
+        val item = if (kills == 0L) Items.GRAY_DYE.defaultInstance
+        else icon.getItem()
+
+        item.withTooltip {
+            add(name)
+
+            add("Kills: ") {
+                color = TextColor.GRAY
+                append(kills.toFormattedString()) { color = TextColor.YELLOW }
+                append("/") { color = TextColor.GOLD }
+                append(requiredKills.toFormattedString()) { color = TextColor.YELLOW }
+                val percentage = kills / requiredKills.toDouble() * 100
+
+                if (percentage >= 100) {
+                    append(" Maxed!") {
+                        color = TextColor.RED
+                    }
+                } else {
+                    append(" (")
+                    append(percentage.round()) {
+                        color = TextColor.GREEN
+                        append("%")
+                    }
+                    append(")")
+                }
+            }
+
+            add("Level: ") {
+                color = TextColor.GRAY
+                append("$currentLevel") { color = TextColor.YELLOW }
```

Comment:
Maybe change the color to red while its not done, kinda like we do it with the essence perks